server/src/main/java/org/elasticsearch/index/fielddata/FieldDataStats.java

```diff
@@ -104,11 +155,93 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         FieldDataStats that = (FieldDataStats) o;
-        return memorySize == that.memorySize && evictions == that.evictions && Objects.equals(fields, that.fields);
+        return memorySize == that.memorySize
+            && evictions == that.evictions
+            && Objects.equals(fields, that.fields)
+            && Objects.equals(globalOrdinalsStats, that.globalOrdinalsStats);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(memorySize, evictions, fields);
+        return Objects.hash(memorySize, evictions, fields, globalOrdinalsStats);
+    }
+
+    public static class GlobalOrdinalsStats {
+
+        private long buildTimeMillis;
+        private final Map<String, GlobalOrdinalFieldStats> fieldGlobalOrdinalsStats;
+
+        public GlobalOrdinalsStats(long buildTimeMillis, Map<String, GlobalOrdinalFieldStats> fieldGlobalOrdinalsStats) {
+            this.buildTimeMillis = buildTimeMillis;
+            this.fieldGlobalOrdinalsStats = Objects.requireNonNull(fieldGlobalOrdinalsStats);
+        }
+
+        public long getBuildTimeMillis() {
+            return buildTimeMillis;
+        }
+
+        public Map<String, GlobalOrdinalFieldStats> getFieldGlobalOrdinalsStats() {
+            return fieldGlobalOrdinalsStats;
+        }
+
+        void add(GlobalOrdinalsStats other) {
+            buildTimeMillis += other.buildTimeMillis;
+            for (var entry : other.fieldGlobalOrdinalsStats.entrySet()) {
+                fieldGlobalOrdinalsStats.merge(
+                    entry.getKey(),
+                    entry.getValue(),
+                    (value1, value2) -> new GlobalOrdinalFieldStats(
+                        value1.totalBuildingTime + value2.totalBuildingTime,
+                        Math.max(value1.valueCount, value2.valueCount)
+                    )
+                );
+            }
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            GlobalOrdinalsStats that = (GlobalOrdinalsStats) o;
+            return buildTimeMillis == that.buildTimeMillis && Objects.equals(fieldGlobalOrdinalsStats, that.fieldGlobalOrdinalsStats);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(buildTimeMillis, fieldGlobalOrdinalsStats);
```

Comment:
I don't think these classes will ever be used in hash map or other data structure that requires proper implementation of equals/hashcode. It just exists for the serialization tests (otherwise comparing expected and actual instances is a cumbersome). Note that this an existing pattern for stats classes like this (e.g. `FieldDataStats`, `IndexingStats`, `DocsStats` etc.). I think adding class level java doc warning makes sense to me.